packages/integration-karma/test/shadow-dom/ShadowRoot.elementsFromPoint/index.spec.js

```diff
@@ -0,0 +1,131 @@
+import { createElement } from 'lwc';
+import Container from 'x/container';
+
+describe('elementsFromPoint', () => {
+    // The browsers disagree on whether elements _above_ the shadow root should also be included
+    // when calling shadowRoot.elementsFromPoint(). Firefox only returns elements inside of the
+    // immediate shadow root, whereas Chrome and Safari return elements above (outside) of that root.
+    // https://crbug.com/1207863#c4
+    let onlyIncludesElementsInImmediateShadowRoot = false;
+    beforeAll(() => {
+        if (process.env.NATIVE_SHADOW) {
+            // detect the Firefox behavior
+            const div = document.createElement('div');
+            div.attachShadow({ mode: 'open' }).innerHTML = '<div>foo</div>';
+            document.body.appendChild(div);
+            const { left, top } = div.shadowRoot.querySelector('div').getBoundingClientRect();
+            const elements = div.shadowRoot.elementsFromPoint(left, top);
+            document.body.removeChild(div);
+            onlyIncludesElementsInImmediateShadowRoot = elements.length === 1;
+        }
+    });
+
+    function test(element, expectedElements) {
+        const { left, top, width, height } = element.getBoundingClientRect();
+        const rootNode = element.getRootNode();
+        const elementsFromPoint = rootNode.elementsFromPoint(left + width / 2, top + height / 2);
+
+        if (onlyIncludesElementsInImmediateShadowRoot) {
+            const mainRootNode = element.getRootNode();
+            expectedElements = expectedElements.filter((el) => el.getRootNode() === mainRootNode);
+        }
+
+        expect(elementsFromPoint).toEqual(expectedElements);
+    }
+
+    // Some browsers have unpredictable behavior for some of these tests
+    const isIE = /(MSIE|Trident)/.test(navigator.userAgent);
+    const isSafari = /Safari/.test(navigator.userAgent) && !/Chrom/.test(navigator.userAgent);
+    const isOldChrome = /Chrom/.test(navigator.userAgent) && process.env.COMPAT;
+    const isOldSafari = isSafari && process.env.COMPAT;
```

Comment:
These browser workarounds are nuts, but I will say I'm comforted that we can run all the tests in Chrome and Firefox, in both synthetic and native mode.

Reply:
I wanted to run the basic test against IE, since otherwise we don't have any scenario where the native `elementsFromPoint` returns null, so we aren't testing the `ifNull()` check anywhere. Do you still think we should remove the test?

And yeah I can move the UA sniffing to the test-utils, no prob.